### PR TITLE
Update README to work with Bitcoin Core 0.16+

### PR DESCRIPTION
**Pull request by Stadicus**
The Bitcoin Core `addwitnessaddress` rpc command is deprecated in version 0.16.0. Thus, the example `bitcoin.conf` needs to be extended to with the option `deprecatedrpc=addwitnessaddress`.

Without this change, Eclair will not be able to open new channels and the following error message will be written to eclair.log: 

```
WARN akka.actor.OneForOneStrategy akka://default/user/$h/switchboard/peer-0390b6fbf04dd277955ce454c87b270d3c910fa96ef3d78afa4bf17d3b733fb877 - addwitnessaddress is deprecated and will be fully removed in v0.17. To use addwitnessaddress in v0.16, restart bitcoind with -deprecatedrpc=addwitnessaddress. Projects should transition to using the address_type argument of getnewaddress, or option -addresstype=[bech32|p2sh-segwit] instead. (code: -32)
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +42,8 @@ rpcpassword=bar
 txindex=1
 zmqpubrawblock=tcp://127.0.0.1:29000
 zmqpubrawtx=tcp://127.0.0.1:29000
+
+deprecatedrpc=addwitnessaddress   # only needed with Bitcoin Core 0.16+
```
Comment (sstone) [inline README.md]: You also need to specify `addresstype=p2sh-segwit` (we don't support bech32 addresses yet).
The sentence below about UTXO management should also be amended: with version 0.16 it is no longer necessary to specify change addresses manually.

Comment (Stadicus) [reply README.md]: According to the Bitcoin Core 0.16.0 release notes, the addresstype "p2sh-segwit" is the default, bech32 is not enforced. 
https://bitcoin.org/en/release/v0.16.0#segwit-wallet

I think the whole paragraph "On testnet..." should be marked for Bitcoin Core 0.15 and earlier, like "On testnet with Bitcoin Core 0.15.1 and earlier...". And a new paragraph for Bitcoin Core 0.16+ should be added. I will test this thoroughly (wiping my wallet and creating a new, 0.16-native wallet) to describe this.

Comment (sstone) [reply README.md]: `p2sh-segwit` is the default setting but if people change it eclair will not work properly, explicitly adding this option to `bitcoin.conf` makes it clearer.

Comment (Stadicus) [reply README.md]: Following your explanation, I added the `addresstype=p2sh-segwit`.

I specified more clearly that the special handling of UTXO as `p2sh-of-p2wpkh` is needed in wallets created with Bitcoin Core 0.15.1 and earlier, as even a wallet migrated to 0.16.0 could contain `legacy` UTXO.

I tested setting up a newly created wallets in 0.16.0 and everything went as expected without any special UTXO handling.

File: README.md [hunk 2]
```diff
@@ -42,15 +42,21 @@ rpcpassword=bar
...
+
+# lines below only needed with Bitcoin Core 0.16+
+deprecatedrpc=addwitnessaddress
+addresstype=p2sh-segwit
 ```
 
 Eclair will use any BTC it finds in the Bitcoin Core wallet to fund any channels you choose to open. Eclair will return BTC from closed channels to this wallet.
 
-On **__testnet__**, you also need to make sure that all your UTXOs are `p2sh-of-p2wpkh`.
+On **__testnet__** and wallets created with Bitcoin Core 0.15.1 and earlier, you also need to make sure that all your UTXOs are `p2sh-of-p2wpkh`.
```
Comment (sstone) [inline README.md]: It is correct but I would phrase it differently: there are 3 options and we should explain what users will have to do in each case:

- fresh installation of Bitcoin Core 0.16: nothing to do
- migration of an existing wallets to 0.16: send all your balance to a new address
- users who stick to 0.15.1: as before

**Pull request merged**